### PR TITLE
fix(#7044): toWei trim strings with more than 20 decimal places

### PR DESCRIPTION
**Pull request by gordon-to**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run lint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

**Comment by gordon-to** [comment]
fixes #7044

File: packages/web3-utils/src/converters.ts [hunk 1]
```diff
@@ -558,21 +558,22 @@ export const toWei = (number: Numbers, unit: EtherUnits): string => {
 	if (!denomination) {
 		throw new InvalidUnitError(unit);
 	}
-	let parsedNumber = number;
-	if (typeof parsedNumber === 'number'){
-		if (parsedNumber  < 1e-15){
+	let parsedNumber: string;
```
Comment (jdevcs) [inline packages/web3-utils/src/converters.ts]: @gordon-to thanks for PR, 

due to this change , there are build errors: 
```
Error: web3-utils: src/converters.ts(580,10): error TS2454: Variable 'parsedNumber' is used before being assigned.
Error: web3-utils: src/converters.ts(580,92): error TS24[54](https://github.com/web3/web3.js/actions/runs/9125740835/job/25095746688?pr=7045#step:6:55): Variable 'parsedNumber' is used before being assigned.
```

**Comment by luu-alex** [comment]
hey there, the PR is failing some of the testcases, in the `web3.js/packages/web3-utils` run yarn test:unit to see tests

**Comment by luu-alex** [comment]
I added more tests and also added a fix for the function `fromWei`. Someone on the team will review this PR
Thanks for your contribution :)

**Review by jdevcs — APPROVED** [review]

**Review by avkos — APPROVED** [review]

**Pull request merged**